query-engine/prisma-models/src/model.rs

```diff
@@ -52,6 +52,17 @@ impl Model {
             .unwrap()
     }
 
+    pub fn supports_create_operation(&self) -> bool {
+        self.supports_create_operation
+    }
+
+    pub fn filter_all_fields<P>(&self, predicate: P) -> Vec<Field>
+    where
+        P: FnMut(&&Field) -> bool,
+    {
+        self.fields().all.iter().filter(predicate).map(Clone::clone).collect()
+    }
```

Comment:
Related to the comment above, I'd prefer to keep this in the `Fields` struct where all field-related methods are located. Something like `model.fields().filter_all(...)`

Reply:
I think this abstraction is correct. `model.fields().filter_all` wouldn't work given that `fields()` returns a vec of the fields from the model. Maybe I need to rename it but this is saying run the predicate filter through all the fields on this model.

Reply:
I guess what we could do is `model.field.filter_all(...)`

Reply:
Hmm, `fields()` doesn't return a `Vec<Field>` https://github.com/prisma/prisma-engines/blob/813d3e391bd044922b873f602de7069527dc78e9/query-engine/prisma-models/src/model.rs#L48-L53

Reply:
Ah sorry. I misunderstood that.